trace/sampling.go

```diff
@@ -73,3 +73,11 @@ func NeverSample() Sampler {
 		return SamplingDecision{Sample: false}
 	}
 }
+
+// AlwaysParentSampler returns a Sampler that only samples spans,
+// and only if the parent span has already decided that it should be sampled.
```

Comment:
nit: It may be better to reword the Doc to
```
AlwaysParentSampler returns a Sampler that samples a span if it's parent span is sampled. The sampler does not sample a root span (a span without a parent).
```